internal/cluster/cluster.go

```diff
@@ -1532,3 +1533,59 @@ func (m *Manager) updateSchedulableMastersForcedTrue(ctx context.Context, cluste
 
 	return nil
 }
+
+func (m *Manager) HandleVerifyVipsResponse(ctx context.Context, clusterID strfmt.UUID, stepReply string) error {
+	log := logutil.FromContext(ctx, m.log)
+	return m.db.Transaction(func(tx *gorm.DB) error {
+		var response models.VerifyVipsResponse
+		if err := json.Unmarshal([]byte(stepReply), &response); err != nil {
+			log.WithError(err).Error("HandleVerifyVipsResponse: unmarshal")
+		}
+		cluster, err := common.GetClusterFromDB(common.LoadTableFromDB(common.LoadTableFromDB(tx, common.APIVIPsTable), common.IngressVIPsTable),
+			clusterID, common.SkipEagerLoading)
+		if err != nil {
+			log.WithError(err).Errorf("HandleVerifyVipsResponse: getting cluster %s", clusterID.String())
+			return err
+		}
```

Comment:
This func is quite large:/ Maybe extract lines 1544-1549 to something like: getClusterWithVIPs

Reply:
OK